### PR TITLE
Add filter argument to Ringbuffer interface [API-1294]

### PR DESCRIPTION
**Pull request by fatihozer0**
I am not sure if anything is right but can not say why it is wrong. I tried this on some other codes to try if it works and worked but not sure.

### CODE REVIEW

**Comment by srknzl** [comment]
> I am not sure if anything is right but can not say why it is wrong. 

😄 What? 

Looks good, thanks

**Review by srknzl — DISMISSED** [review]

**Review by yuce — CHANGES_REQUESTED** [review]
Could you port Java client's tests and add documentation and a code sample?

**Comment by srknzl** [comment]
Adding a test may not be possible because server side code is needed. I don't know if we have a filter in the test jar. A code sample requires users to write java code in the server side. Not good.

You are right about the documentation. API documentation can be added.  Also there can be a small section about the filter in section https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/DOCUMENTATION.md#847-using-ringbuffer

**Comment by yuce** [comment]
@srknzl The python client has two tests which uses the filter: https://github.com/hazelcast/hazelcast-python-client/blob/b8dd0d737ef976922311a8241e2a51b50e7ab3fa/tests/integration/backward_compatible/proxy/ringbuffer_test.py#L230 Could you guys check out them to see whether they can be ported to the Node.js client?

**Comment by srknzl** [comment]
We can do and we should. It is in the test jar. 

https://github.com/hazelcast/hazelcast/blob/14ef0dbefa4a6fa00a50a8873aeea02ec6c3f5e4/hazelcast/src/test/java/com/hazelcast/client/test/PrefixFilter.java#L26

**Comment by fatihozer0** [comment]
Added a test which is not ported. I checked the links you shared. Gonna add documentations now

File: src/proxy/Ringbuffer.ts [hunk 1]
```diff
@@ -164,5 +164,5 @@ export interface Ringbuffer<E> extends DistributedObject {
      *                      or if `minCount` larger than `maxCount`,
      *                      or if `maxCount` larger than `1000` (to prevent overloading)
      */
-    readMany(startSequence: number | Long, minCount: number, maxCount: number): Promise<ReadResultSet<E>>;
```
Comment (srknzl) [inline src/proxy/Ringbuffer.ts:167]: Can you add the API documentation?

Comment (fatihozer0) [reply src/proxy/Ringbuffer.ts:167]: Done.

**Review by srknzl — APPROVED** [review]

**Review by yuce — APPROVED** [review]

**Comment by fatihozer0** [comment]
Continues on #1413

**Pull request merged**